OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -96,10 +105,12 @@ public class Bullet : IProjectile, ISync
 		ContrailRenderable contrail;
 		string trailPalette;
 
-		[Sync] WPos pos, target;
+		[Sync] WPos pos, target, source;
 		int length;
+		int distance;
 		[Sync] int facing;
 		int ticks, smokeTicks;
+		int bounceCount;
```

Comment:
The `BounceCount` name in the `BulletInfo` is fine, because in that context it is clear that is the total/maximum number of bounces.